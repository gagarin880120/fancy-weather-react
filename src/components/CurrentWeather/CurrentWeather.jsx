import React from 'react';
import styles from './CurrentWeather.module.css';

export default function CurrentWeather() {
  return (
    <div className={styles.wrapper}>
      <div className={styles.temperature}>
        20Â°
      </div>
      <div className={styles.humidity}>
        50%
      </div>
      <div className={styles.wind}>
        3 m/s
      </div>
    </div>
  );
}
