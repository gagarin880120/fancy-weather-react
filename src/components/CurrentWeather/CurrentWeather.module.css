.wrapper {
  display: flex;
}

.temperature {
  position: relative;
  font-size: 196px;
  font-weight: bold;
}

.details {
  margin-top: 130px;
  font-size: 18px;
}

.icon {
  position: absolute;
  top: -20px;
  right: -170px;
}

@media screen and (max-width: 460px) {
  .wrapper {
    flex-direction: column;
  }

  .temperature {
    font-size: 120px;
  }

  .icon {
    top: -40px;
    left: 145px;
    width: 100px;
  }

  .details {
    margin: 10px 0 20px;
  }
}
