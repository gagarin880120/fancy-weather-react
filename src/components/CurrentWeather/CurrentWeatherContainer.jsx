import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import CurrentWeather from './CurrentWeather';
import { selectCurrentWeather } from '../../redux/selectors';

export function CurrentWeatherContainer({ currentWeather }) {
  // console.log(currentWeather);
  return (
    currentWeather
      ? (
        <CurrentWeather
          currentWeather={currentWeather}
        />
      ) : <div>Loading...</div>
  );
}

export const mapStateToProps = (state) => ({
  currentWeather: selectCurrentWeather(state),
});

CurrentWeatherContainer.propTypes = {
  currentWeather: PropTypes.shape({}),
};

CurrentWeatherContainer.defaultProps = {
  currentWeather: {},
};

export default connect(mapStateToProps)(CurrentWeatherContainer);
