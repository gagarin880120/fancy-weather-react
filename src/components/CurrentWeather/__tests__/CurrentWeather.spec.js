import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import CurrentWeather from '../CurrentWeather';

jest.mock('@intern0t/react-weather-icons', () => ({sunny: jest.fn()}));

describe('CurrentWeather', () => {
  describe('component', () => {
    test('should render without crashing', () => {
      const renderer = new ShallowRenderer();
      renderer.render(
        <CurrentWeather
          currentWeather={{
            icon: 'sunny',
            weather: {
              description: 'Sunny weather'
            }
          }}
        />
      );

      const result = renderer.getRenderOutput();

      expect(result).toMatchSnapshot();
    });
  });
});
