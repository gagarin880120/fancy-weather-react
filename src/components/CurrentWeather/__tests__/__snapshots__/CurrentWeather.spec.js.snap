// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentWeather component should render without crashing 1`] = `
<div
  className="wrapper"
>
  <div
    className="temperature"
  >
    <mockConstructor
      className="icon"
      color="#fff"
      size={170}
    />
  </div>
  <div
    className="details"
  >
    <div
      className="summary"
    >
      Sunny weather
    </div>
    <div
      className="humidity"
    />
    <div
      className="wind"
    />
  </div>
</div>
`;
